src/data/indicator_function.lean

```diff
@@ -150,9 +164,30 @@ end
 
 end add_group
 
+section mul_zero_class
+variables [mul_zero_class β] {s t : set α} {f g : α → β} {a : α}
+
+lemma indicator_mul (s : set α) (f g : α → β) :
+  indicator s (λa, f a * g a) = λa, indicator s f a * indicator s g a :=
+by { funext, simp only [indicator], split_ifs, { refl }, rw mul_zero }
+
+end mul_zero_class
+
 section order
 variables [has_zero β] [preorder β] {s t : set α} {f g : α → β} {a : α}
 
+lemma indicator_nonneg (h : ∀ a ∈ s, 0 ≤ f a) : ∀ a, 0 ≤ indicator s f a :=
+by { assume a, rw indicator_apply, split_ifs with as, { exact h _ as }, refl }
+
+lemma indicator_nonneg' (h : a ∈ s → 0 ≤ f a) : 0 ≤ indicator s f a :=
+by { rw indicator_apply, split_ifs with as, { exact h as }, refl }
```

Comment:
I'd prove this one first, then `lemma indicator_nonneg ... := λ a, indicator_nonneg' (h a)`. Same for `nonpos`.